ui/appui-react/src/appui-react/frontstage/FrontstageDef.tsx

```diff
@@ -1086,3 +1142,37 @@ export class FrontstageDef {
     };
   }
 }
+
+function createWidgetDef(config: WidgetConfig | undefined, stableId: string, type: WidgetType): WidgetDef | undefined {
+  if (!config)
+    return undefined;
+
+  const stableConfig = getStableWidgetProps(config, stableId);
+  const widgetDef = new WidgetDef(stableConfig);
+  widgetDef.initializeFromConfig(stableConfig, type);
```

Comment:
does this seem unecessary? why not just pass the type to the constructor and encapsulate all of this?
same goes for the panel